examples/vision/mixUp.py

```diff
@@ -0,0 +1,225 @@
+"""
+Title: MixUp augmentation for image classification
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/03/06
+Last modified: 2021/03/06
+Description: Data augmentation using the mixup technique for image classification.
+"""
+"""
+## Introduction
+"""
+
+"""
+_mixup_ is a *domain-agnostic* data augmentation technique proposed in [mixup: Beyond Empirical Risk Minimization](https://arxiv.org/abs/1710.09412) 
+by Zhang et al. It's implemented with the following formulas:
+
+![](https://i.ibb.co/DRyHYww/image.png)
+
+(Note that the lambda values are values with the [0, 1] range and are sampled from the
+[Beta distribution](https://en.wikipedia.org/wiki/Beta_distribution).)
+
+The technique is quite systematically named - we are literally mixing up the features and
+their corresponding labels. Implementation-wise it's simple. Neural networks are prone
+to [memorizing corrupt labels](https://arxiv.org/abs/1611.03530). mixup relaxes this by
+combining different features with one another (same happens for the labels too) so that
+a network does not get overconfident about the relationship between the features and
+their labels. 
+
+mixup is specifically useful when we are not sure about selecting a set of augmentation
+transforms for a given dataset, medical imaging datasets, for example. mixup can be
+extended to a variety of data modalities such as computer vision, naturallanguage
+processing, speech, and so on.
+
+This example requires TensorFlow 2.4 or higher.
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from tensorflow.keras import layers
+
+tf_distribution = tf.compat.v1.distributions
```

Comment:
We should also not depend on deprecated functionality.

Looking at the source, here's how it's implemented:

```python
  def _sample_n(self, n, seed=None):
    expanded_concentration1 = array_ops.ones_like(
        self.total_concentration, dtype=self.dtype) * self.concentration1
    expanded_concentration0 = array_ops.ones_like(
        self.total_concentration, dtype=self.dtype) * self.concentration0
    gamma1_sample = random_ops.random_gamma(
        shape=[n],
        alpha=expanded_concentration1,
        dtype=self.dtype,
        seed=seed)
    gamma2_sample = random_ops.random_gamma(
        shape=[n],
        alpha=expanded_concentration0,
        dtype=self.dtype,
        seed=distribution_util.gen_new_seed(seed, "beta"))
    beta_sample = gamma1_sample / (gamma1_sample + gamma2_sample)
    return beta_sample
```

This is easy to reimplement from scratch:

```
def sample_beta_distribution(size, concentration_0=0.2, concentration_1=0.2):
  gamma_1_sample = tf.random.gamma(
      shape=[size],
      alpha=concentration_1)
  gamma_2_sample = tf.random.gamma(
      shape=[size],
      alpha=concentration_0)
  return gamma_1_sample / (gamma_1_sample + gamma_2_sample)
```

Unless I made any copy/paste mistake (better to double check).

Reply:
Thank you! I have checked it and incorporated it inside the code.